### PR TITLE
Add reader for FY-3 VIRR

### PR DESCRIPTION
**Pull request by wroberts4**
Add a FY3 virr reader that supports 7 reflective (visible) channels and 3 emissive (IR) channels. True color and Rayleigh corrections are also included. This feature was requested by Liam E. Gumley, Kathleen Strabala, and David Hoese.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed 
 - [ ] Passes ``git diff origin/master -- "*py" | flake8 --diff`` 
 - [ ] Fully documented

### CODE REVIEW

File: satpy/readers/virr_l1b.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def __init__(self, filename, filename_info, filetype_info):
+        super(VIRR_L1B, self).__init__(filename, filename_info, filetype_info)
+        logging.debug('day/night flag for {0}: {1}'.format(filename, self['/attr/Day Or Night Flag']))
+        self.geolocation_prefix = filetype_info['geolocation_prefix']
+        self.platform_id = filename_info['platform_id']
+        self.l1b_prefix = 'Data/'
+        self.wave_number = 'Emissive_Centroid_Wave_Number'
+        if filename_info['platform_id'] == 'FY3B':
+            self.l1b_prefix = ''
+            self.wave_number = 'Emmisive_Centroid_Wave_Number'
```
Comment (djhoese) [inline satpy/readers/virr_l1b.py]: Is this a typo in the code or in the file? Isn't it "Emissive"?

File: satpy/tests/reader_tests/test_virr_l1b.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        self.assertEqual(units, attributes['units'])
+        self.assertEqual(calibration, attributes['calibration'])
+        self.assertEqual(standard_name, attributes['standard_name'])
+        self.assertEqual(file_type, attributes['file_type'])
+        self.assertTrue(attributes['band_index'] in range(band_index_size))
+        self.assertEqual(resolution, attributes['resolution'])
+        self.assertEqual(level, attributes['level'])
+        self.assertEqual(('longitude', 'latitude'), attributes['coordinates'])
+
+    def _FY3_helper(self, platform_name, reader, Emissive_units):
```
Comment (djhoese) [inline satpy/tests/reader_tests/test_virr_l1b.py]: I'd prefer if these methods were all lowercase even though these are acronyms.

File: satpy/etc/composites/virr.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+sensor_name: visir/virr
+
+modifiers:
+  sunz_corrected:
+    compositor: !!python/name:satpy.composites.SunZenithCorrector
+    prerequisites:
+    - name: solar_zenith_angle
```
Comment (djhoese) [inline satpy/etc/composites/virr.yaml]: Does `sunz_corrected` need to be defined? Isn't it defined and used from `visir.yaml`?

File: satpy/readers/virr_l1b.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+from datetime import datetime
```
Comment (djhoese) [inline satpy/readers/virr_l1b.py]: Every python module should copy the header that all other files have. We're still standardizing everything but the copyright author should be "SatPy developers" I think. Blank lines in the header should have `#` also. If you can include a module docstring that would be for the best (describe the file format, what satellites are supported, etc).

File: satpy/readers/virr_l1b.py (hunk 1 above)
Comment (wroberts4) [reply satpy/readers/virr_l1b.py]: Both the file and the documents you sent me have "Emissive". Here's a quote from the document you sent me awhile back: "Note the unusual spelling in Emmisive_Centroid_Wave_Number for VIRR " at the bottom of page 5.

Comment (djhoese) [reply satpy/readers/virr_l1b.py]: What does `ncdump -h` say?

File: satpy/etc/composites/virr.yaml (hunk 3 above)
Comment (wroberts4) [reply satpy/etc/composites/virr.yaml]: `sunz_corrected` is defined differently in visir.yaml: visir.yaml uses "optional_prerequisites" and virr.yaml uses "prerequisites". I'm not sure if there's a better way to specify that without copying the few extra lines of code over. On top of that I don't know how virrs.yaml would use visir.yaml's `sunz_corrected`.

Comment (djhoese) [reply satpy/etc/composites/virr.yaml]: Ah good point. This is fine by me then.

File: satpy/readers/virr_l1b.py (hunk 1 above)
Comment (wroberts4) [reply satpy/readers/virr_l1b.py]: "Emmisive_Centroid_Wave_Number"

**Pull request merged**